src/internal/connector/exchange/api/contacts.go

```diff
@@ -190,95 +190,140 @@ func (c Contacts) EnumerateContainers(
 var _ itemPager = &contactPager{}
 
 type contactPager struct {
-	gs      graph.Servicer
-	builder *users.ItemContactFoldersItemContactsDeltaRequestBuilder
-	options *users.ItemContactFoldersItemContactsDeltaRequestBuilderGetRequestConfiguration
+	gs          graph.Servicer
+	user        string
+	directoryID string
+
+	// switch between using delta vs non-delta fetch
+	nonDelta bool
+
+	builder *users.ItemContactFoldersItemContactsRequestBuilder
+	options *users.ItemContactFoldersItemContactsRequestBuilderGetRequestConfiguration
+
+	deltaBuilder *users.ItemContactFoldersItemContactsDeltaRequestBuilder
+	deltaOptions *users.ItemContactFoldersItemContactsDeltaRequestBuilderGetRequestConfiguration
 }
 
-func (p *contactPager) getPage(ctx context.Context) (api.DeltaPageLinker, error) {
-	resp, err := p.builder.Get(ctx, p.options)
+func NewContactPager(
+	gs graph.Servicer,
+	user, directoryID, deltaURL string,
+	immutableIDs bool,
+) (contactPager, error) {
+	options, err := optionsForContactFoldersItem([]string{"parentFolderId"}, immutableIDs)
 	if err != nil {
-		return nil, graph.Stack(ctx, err)
+		return contactPager{}, err
 	}
 
-	return resp, nil
-}
+	deltaOptions, err := optionsForContactFoldersItemDelta([]string{"parentFolderId"}, immutableIDs)
+	if err != nil {
+		return contactPager{}, err
+	}
 
-func (p *contactPager) setNext(nextLink string) {
-	p.builder = users.NewItemContactFoldersItemContactsDeltaRequestBuilder(nextLink, p.gs.Adapter())
-}
+	deltaBuilder := gs.Client().UsersById(user).ContactFoldersById(directoryID).Contacts().Delta()
+	if deltaURL != "" {
+		deltaBuilder = users.NewItemContactFoldersItemContactsDeltaRequestBuilder(deltaURL, gs.Adapter())
+	}
+
+	return contactPager{
+		gs:          gs,
+		user:        user,
+		directoryID: directoryID,
+		nonDelta:    false,
```

Comment:
We could add an option  to start with non delta, but we already know the behaviour of delta when mailbox is full and it will anyways take us one extra request either way.